pytential/symbolic/pde/system_utils.py

```diff
@@ -0,0 +1,529 @@
+            knl = source_kernel
+            for axis_var, nrepeats in derivatives:
+                axis = ds.index(axis_var)
+                for _ in range(nrepeats):
+                    knl = AxisSourceDerivative(axis, knl)
+            new_source_kernels.append(knl)
+        new_int_g = int_g.copy(source_kernels=new_source_kernels)
+
+        (monom, coeff,) = rest_terms.terms()[0]
+        # Now from d[0]*y[1], we separate the two terms
+        # d terms end up in the expression and y terms end up in the density
+        d_terms, y_terms = monom[:len(ds)], monom[len(ds):]
+        expr_multiplier = _monom_to_expr(d_terms, ds)
+        density_multiplier = _monom_to_expr(y_terms, sources_pymbolic) \
+                * conv(coeff)
+        # since d/d(d) = - d/d(y), we multiply by -1 to get source derivatives
+        density_multiplier *= (-1)**int(sum(nrepeats for _, nrepeats in derivatives))
+        new_int_gs = _multiply_int_g(new_int_g, sym.sympify(expr_multiplier),
+                density_multiplier)
+        result.extend(new_int_gs)
+    return result
+
+
+def _multiply_int_g(int_g: IntG, expr_multiplier: sym.Basic,
+        density_multiplier: ExpressionT) -> List[IntG]:
+    """Multiply the expression in ``IntG`` with the *expr_multiplier*
+    which is a symbolic expression and multiply the densities
+    with *density_multiplier* which is a pymbolic expression.
+    """
+    from sumpy.symbolic import SympyToPymbolicMapper
+    result = []
+
+    base_kernel = int_g.target_kernel.get_base_kernel()
+    sym_d = make_sym_vector("d", base_kernel.dim)
+    base_kernel_expr = _get_kernel_expression(base_kernel.expression,
+            int_g.kernel_arguments)
+    subst = {pymbolic.var(f"d{i}"): pymbolic.var("d")[i] for i in
+            range(base_kernel.dim)}
+    conv = SympyToPymbolicMapper()
+
+    if expr_multiplier == 1:
+        # if there's no expr_multiplier, only multiply the densities
+        return [int_g.copy(densities=tuple(density*density_multiplier
+            for density in int_g.densities))]
+
+    for knl, density in zip(int_g.source_kernels, int_g.densities):
+        if expr_multiplier == 1:
+            new_knl = knl.get_base_kernel()
+        else:
+            new_expr = conv(knl.postprocess_at_source(base_kernel_expr, sym_d)
+                    * expr_multiplier)
+            new_expr = pymbolic.substitute(new_expr, subst)
+            new_knl = ExpressionKernel(knl.dim, new_expr,
+                knl.get_base_kernel().global_scaling_const,
+                knl.is_complex_valued)
+        result.append(int_g.copy(target_kernel=new_knl,
+            densities=(density*density_multiplier,),
+            source_kernels=(new_knl,)))
+    return result
+
+
+def convert_int_g_to_base(int_g: IntG, base_kernel: ExpressionKernel) \
+        -> ExpressionT:
+    """Converts an *IntG* to an expression with *IntG*s having the
+    base kernel *base_kernel*.
+    """
```

Comment:
From looking at `_convert_int_g_to_base` below, it seems more like this replaces the kernel, so maybe `replace_int_g_base_kernel`?

Reply:
Fixed in f7b661de28cbcea9fc29cd122a128e7111353210